### PR TITLE
2018 Sonata 2.0T Limited Fingerprint

### PR DESCRIPTION
**Pull request by Felixowns**
Makes the 2018 Sonata identify properly and not require multiple ecu power cycles for comma to detect vehicle. Properly identifies as the sonata LF (2018-2019 are the same car) instead of previously identifying as "KIA OPTIMA SX 2019 & 2016" which worked but took multiple power cycles before openpilot would enable. Did a quick road test but will do a longer one to see if any other vehicle specific parameters need to be adjusted.

Dongle Id:0481820f6e50c2b7

### CODE REVIEW

**Comment by pd0wm** [comment]
No need to add legacy fingerprint. Can you also fix the whitepsace?

**Comment by pd0wm** [comment]
Looks like your car only returns a fw version for the `fwdCamera`, which is very odd. Any idea what might cause that? All wires connected properly?

I removed the legacy CAN fingerprint since we don't want to touch those anymore, but without it won't work for you it seems.

**Comment by Felixowns** [comment]
I'll take a peek at the different drives in a bit to see. There were a couple on 8/1 where I was testing the changes. First I just did the v2 fingerprint but then OP said warning, closest match found 2019 Hyundai Sonata. Then when I added V1 it works normally. If you pull up any older drives where I actually had OP engaged that might give the info? I hadn't changed anything in the car and all the wires seem attached fine. Only temporarily did I have the comma unplugged when I was setting up ssh on 8/1.

**Comment by Felixowns** [comment]
As for those lines not lining up thats weird. Was fine in my editor but when github desktop uploaded it did that. Very strange, I'll try to make sure it doesn't do that next time.

**Comment by Felixowns** [comment]
So I checked my most recent drive to work today and yea same thing where only fwdCamera was showing the fwVersion. Everything work fine though. And the only change I did on my comma was the fingerprinting so I don't know why the old drives had the data but not the new ones. When I go home I'll try to power cycle the car a few times and see if I get different computers to detect again.

**Comment by Felixowns** [comment]
I believe I fixed the fw detection issue. I double checked the harness, may have been a loose wire or something. Detecting just fine now using v2 fingerprint only.

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -283,16 +283,21 @@ class Buttons:
       b'\xf1\x00LF__ SCC F-CUP      1.00 1.00 96401-C2200         ',
     ],
     (Ecu.esp, 0x7d1, None): [
+      b'\xf1\000LF ESC \f 11 \027\001\023 58920-C2610',
```
Comment (pd0wm) [inline selfdrive/car/hyundai/values.py:286]:
```suggestion
      b'\xf1\x00LF ESC \f 11 \x17\x01\x13 58920-C2610',
```

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -283,16 +283,21 @@ class Buttons:
...
       b'\xf1\x00LF ESC \t 11 \x17\x01\x13 58920-C2610',
     ],
     (Ecu.engine, 0x7e0, None): [
+      b'\xf1\x81606D5051\000\000\000\000\000\000\000\000',
```
Comment (pd0wm) [inline selfdrive/car/hyundai/values.py:290]:
```suggestion
      b'\xf1\x81606D5051\x00\x00\x00\x00\x00\x00\x00\x00',
```

File: selfdrive/car/hyundai/values.py [hunk 3]
```diff
@@ -283,16 +283,21 @@ class Buttons:
...
       b'\xf1\x81606D5K51\x00\x00\x00\x00\x00\x00\x00\x00',
       b'\xf1\x81606G1051\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
     (Ecu.fwdCamera, 0x7c4, None): [
       b'\xf1\x00LFF LKAS AT USA LHD 1.00 1.01 95740-C1000 E51',
+      b'\xf1\xa01.01',
     ],
     (Ecu.transmission, 0x7e1, None): [
+      b'\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\177\xff3\000\xf1\x816W3B1051\000\000\xf1\0006W351_C2\000\0006W3B1051\000\000TLF0T20NL2\000\000\000\000',
```
Comment (pd0wm) [inline selfdrive/car/hyundai/values.py:299]:
```suggestion
      b'\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\x7f\xff3\x00\xf1\x816W3B1051\x00\x00\xf1\x006W351_C2\x00\x006W3B1051\x00\x00TLF0T20NL2\x00\x00\x00\x00',
```

File: selfdrive/car/hyundai/values.py [hunk 4]
```diff
@@ -283,16 +283,21 @@ class Buttons:
...
       b'\xf1\x00LF ESC \t 11 \x17\x01\x13 58920-C2610',
     ],
     (Ecu.engine, 0x7e0, None): [
+      b'\xf1\x81606D5051\000\000\000\000\000\000\000\000',
       b'\xf1\x81606D5K51\x00\x00\x00\x00\x00\x00\x00\x00',
       b'\xf1\x81606G1051\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
     (Ecu.fwdCamera, 0x7c4, None): [
       b'\xf1\x00LFF LKAS AT USA LHD 1.00 1.01 95740-C1000 E51',
+      b'\xf1\xa01.01',
```
Comment (pd0wm) [inline selfdrive/car/hyundai/values.py:296]:
```suggestion
```

**Comment by Felixowns** [comment]
Is there anything I need to do to have the change incorporated in the next release? (I'm a bit of a github noob)

**Comment by Felixowns** [comment]
Hmm so something isn't working correctly now. When I updated OP to the current version I get a close match message but not all the ECUs Fw is being detected
Dongle: 0481820f6e50c2b7
Route: 0481820f6e50c2b7|2021-09-22--23-36-33

      ( ecu = fwdCamera, fwVersion = "\xf1\xa01.01", address = 1988, subAddress = 0),

      ( ecu = engine,
        fwVersion = "\xf1\x81606D5051\000\000\000\000\000\000\000\000",
        address = 2016,
        subAddress = 0 ),

      ( ecu = esp,
        fwVersion = "\xf1\000LF ESC \f 11 \027\001\023 58920-C2610",
        address = 2001,
        subAddress = 0 ),

      ( ecu = transmission,
        fwVersion = "\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\177\xff3\000\xf1\x816W3B1051\000\000\xf1\0006W351_C2\000\0006W3B1051\000\000TLF0T20NL2H\r\xbdm",
        address = 2017,
        subAddress = 0 ) ],


But if I put in my previous car fw (the following) I get a exact match just fine and all fw is detected as expected.

      ( ecu = fwdCamera,
        fwVersion = "\xf1\000LFF LKAS AT USA LHD 1.00 1.01 95740-C1000 E51",
        address = 1988,
        subAddress = 0 ),
      ( ecu = transmission,
        fwVersion = "\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\177\xff3\000\xf1\x816W3B1051\000\000\xf1\0006W351_C2\000\0006W3B1051\000\000TLF0T20NL2\000\000\000\000",
        address = 2017,
        subAddress = 0 ),
      ( ecu = esp,
        fwVersion = "\xf1\000LF ESC \f 11 \027\001\023 58920-C2610",
        address = 2001,
        subAddress = 0 ),
      ( ecu = fwdRadar,
        fwVersion = "\xf1\000LF__ SCC F-CUP      1.00 1.00 96401-C2200         ",
        address = 2000,
        subAddress = 0 ),
      ( ecu = engine,
        fwVersion = "\xf1\x81606D5051\000\000\000\000\000\000\000\000",
        address = 2016,
        subAddress = 0 ) ],

Any idea? should I just resubmit my changes with the exact fw matching.

**Comment by pd0wm** [comment]
Does it work correctly most of the time? You're probably running into https://github.com/commaai/openpilot/issues/21333

**Comment by Felixowns** [comment]
as of 9/20 I had 

      ( ecu = fwdCamera,
        fwVersion = "\xf1\000LFF LKAS AT USA LHD 1.00 1.01 95740-C1000 E51",
        address = 1988,
        subAddress = 0 ),
      ( ecu = transmission,
        fwVersion = "\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\177\xff3\000\xf1\x816W3B1051\000\000\xf1\0006W351_C2\000\0006W3B1051\000\000TLF0T20NL2H\r\xbdm",
        address = 2017,
        subAddress = 0 ),
      ( ecu = esp,
        fwVersion = "\xf1\000LF ESC \f 11 \027\001\023 58920-C2610",
        address = 2001,
        subAddress = 0 ),
      ( ecu = fwdRadar,
        fwVersion = "\xf1\000LF__ SCC F-CUP      1.00 1.00 96401-C2200         ",
        address = 2000,
        subAddress = 0 ),
      ( ecu = engine,
        fwVersion = "\xf1\x81606D5051\000\000\000\000\000\000\000\000",
        address = 2016,
        subAddress = 0 ) ],

and then 9/22 ( I believe after I got the update) I got the close match with only the following constantly appearing (4 drives)

      (ecu = fwdCamera, fwVersion = "\xf1\xa01.01", address = 1988, subAddress = 0),
      ( ecu = engine,
        fwVersion = "\xf1\x81606D5051\000\000\000\000\000\000\000\000",
        address = 2016,
        subAddress = 0 ),
      ( ecu = esp,
        fwVersion = "\xf1\000LF ESC \f 11 \027\001\023 58920-C2610",
        address = 2001,
        subAddress = 0 ),
      ( ecu = transmission,
        fwVersion = "\xf1\x87LAHSGN012918KF10\x98\x88x\x87\x88\x88x\x87\x88\x88\x98\x88\x87w\x88w\x88\x88\x98\x886o\xf6\xff\x98w\177\xff3\000\xf1\x816W3B1051\000\000\xf1\0006W351_C2\000\0006W3B1051\000\000TLF0T20NL2H\r\xbdm",
        address = 2017,
        subAddress = 0 ) ],

I can edit the fingerprint to the old one and it will probably detect all the FWs again. I don't know why it decides to show them only once the finger print exists. but current OP is I get the no exact match every time.

**Comment by pd0wm** [comment]
Feel free to make a new PR with routes!

**Pull request merged**